src/Common/tests/Collections/IDictionaryTest.cs

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Tests.Collections
+{
+    public abstract class IDictionaryTest<TKey, TValue> : ICollectionTest<KeyValuePair<TKey, TValue>>
+    {
+        protected IDictionaryTest(bool isSynchronized)
+            : base(isSynchronized)
+        {
+        }
+
+        protected IDictionary<TKey, TValue> GetDictionary(object[] items)
+        {
+            return (IDictionary<TKey, TValue>)GetCollection(items);
+        }
+
+        [Theory]
+        [InlineData(1)]
+        [InlineData(16)]
+        [InlineData(100)]
+        public void KeysShouldBeCorrect(int count)
+        {
+            object[] items = GenerateItems(count);
+            IDictionary<TKey, TValue> dict = GetDictionary(items);
+            var expectedKeys = new TKey[items.Length];
+            for (int i = 0; i < items.Length; ++i)
+            {
+                expectedKeys[i] = ((KeyValuePair<TKey, TValue>)items[i]).Key;
+            }
+
+            CollectionAssert.Equal(expectedKeys, dict.Keys);
```

Comment:
Is this different from Assert.Equal that takes two IEnumerables? e.g. does this not care about ordering?